src/views/app/index.js

```diff
@@ -50,54 +54,95 @@ const themes = {
 };
 
 export const App = () => {
-    const { darkMode } = useSelector((state) => ({
+    const {
+        darkMode,
+        isPinVerified,
+        isPinEnabled,
+        isPinVerifying,
+    } = useSelector((state) => ({
         darkMode: state.settings.darkMode,
+        existingPin: state.pinConfig.pin,
+        isPinVerified: state.pinConfig.isVerified,
+        isPinEnabled: state.pinConfig.isEnabled,
+        isPinVerifying: state.pinConfig.isVerifying,
     }));
 
+    const dispatch = useDispatch();
+    const appState = useRef(AppState.currentState);
     const [theme, setTheme] = useState(themes.light);
 
+    useEffect(() => {
+        AppState.addEventListener('change', handleAppStateChange);
+
+        return () => {
+            AppState.removeEventListener('change', handleAppStateChange);
+        };
+    }, []);
+
     useEffect(() => {
         SplashScreen.hide();
+        dispatch(triggerPinVerify());
```

Comment:
Probably not needed, since it's already called in `handleAppStateChange`. I'll check later